### PR TITLE
PowerShell  script analyzer and CI

### PR DESCRIPTION
**Pull request by valentin-p**
- Write powershell-script-analyzer.ps1 script to analyze and display the messages
- powershell-script-analyzer.ps1 throws if there is any messages, it succeed only if all rules are respected
- powershell-script-analyzer.ps1 can be executed locally
- Add CI yml file to execute the powershell-script-analyzer.ps1 file

### CODE REVIEW

**Comment by valentin-p** [comment]
Closes #1492

File: .github/workflows/pssanalyzer.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Run PSScriptAnalyzer
+
+on: [push, pull_request]
```
Comment (ErikSchierboom) [inline .github/workflows/pssanalyzer.yml:3]: While not very likely to be used, let's allow the workflow to run manually:

```suggestion
on: [push, pull_request, workflow_dispatch]
```

File: .github/workflows/pssanalyzer.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+jobs:
+  analyze:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # 2.3.4
+      - run: pwsh ./powershell-script-analyzer.ps1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (ErikSchierboom) [inline .github/workflows/pssanalyzer.yml:12]: I don't think you actually use the token, do you?

File: powershell-script-analyzer.ps1 [hunk 3]
```diff
@@ -0,0 +1,17 @@
+if (!(Get-Module -ListAvailable -Name PSScriptAnalyzer)) {
```
Comment (ErikSchierboom) [inline powershell-script-analyzer.ps1:1]: Could you add a synopsis comment section like in the other powershell files?

File: powershell-script-analyzer.ps1 [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+$DETAILS = ($PSSAResults | ForEach-Object { "{0,-11} [{1,-3}, {2,-3}] {3,-50} {4,-35} - {5}" -f $_.Severity, $_.Line, $_.Column, $_.ScriptName, $_.RuleName, $_.Message } | Out-String -Width 88).Trim()
+$DETAILS
+$SUMMARY = ($PSSAResults | Group-Object -Property Severity -NoElement | Foreach-Object { "$($_.Count) $($_.Name)" }) -join ", "
+$SUMMARY
+
+if ($SUMMARY.Count -ne 0) {
+    throw
+}
+
+exit $LastExitCode
```
Comment (ErikSchierboom) [inline powershell-script-analyzer.ps1:17]: Maybe simplify this to:

```suggestion
exit $SUMMARY.Count
```

I don't really like the output of the `throw` :)

**Review by ErikSchierboom — APPROVED** [review]
Excellent! I've left some comments.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**